frontend/components/SettingsPage/SettingsDrawer.tsx

```diff
@@ -0,0 +1,214 @@
+      render: (_: unknown, record: TableData) => (
+        <Flex align="center" gap={8}>
+          <Image
+            src={`/chains/${kebabCase(record.chain)}-chain.png`}
+            alt={record.chain}
+            width={20}
+            height={20}
+            className="flex"
+          />
+          <Text>{record.chain}</Text>
+        </Flex>
+      ),
+      width: '33%',
+    },
+    {
+      title: 'Threshold',
+      key: 'threshold',
+      render: (_: unknown, record: TableData) => (
+        <Flex align="center" gap={8}>
+          <Image
+            src={
+              TokenSymbolConfigMap[
+                record.tokenSymbol as keyof typeof TokenSymbolConfigMap
+              ].image
+            }
+            alt={record.tokenSymbol}
+            width={20}
+            height={20}
+            className="flex"
+          />
+          <Text>{record.threshold}</Text>
+        </Flex>
+      ),
+      width: '33%',
+    },
+    {
+      title: 'Top-up Amount',
+      key: 'topUpAmount',
+      render: (_: unknown, record: TableData) => (
+        <Flex align="center" gap={8}>
+          <Image
+            src={
+              TokenSymbolConfigMap[
+                record.tokenSymbol as keyof typeof TokenSymbolConfigMap
+              ].image
+            }
+            alt={record.tokenSymbol}
+            width={20}
+            height={20}
+            className="flex"
+          />
+          <Text>{record.topUpAmount}</Text>
+        </Flex>
+      ),
+      width: '34%',
+    },
+  ];
+
+  return (
+    <Drawer
+      title={SettingsDrawerTitle}
+      maskClosable
+      closable={false}
+      open={isDrawerOpen}
+      onClose={onClose}
+      width={520}
+      extra={<Button type="text" icon={<CloseOutlined />} onClick={onClose} />}
+    >
+      <Image
+        src="/arrow-split-icon.png"
+        alt="auto-funding"
+        width={36}
+        height={36}
+        className="mb-12"
+      />
+      <Flex gap={26} vertical>
+        <Flex vertical gap={6}>
+          <Text strong>Auto-Funding – Pearl Wallet</Text>
+          <Text className="text-sm">
+            Pearl uses a separate wallet to sign transactions for the Pearl
+            wallet. When balances on the signer wallet fall below the threshold,
+            Pearl refills with the top-up amount from Pearl wallet. This way,
+            your Pearl always stays operational without your action required.
+          </Text>
+        </Flex>
+        <Flex gap={16} vertical>
+          <Flex vertical gap={4}>
+            <Text className="text-xs">Pearl Wallet Address</Text>
+            {masterSafe ? (
+              <AddressLink
+                truncate={false}
+                address={masterSafe?.address}
+                middlewareChain={selectedAgentConfig.middlewareHomeChainId}
+              />
+            ) : (
+              <Text className="text-sm" type="secondary">
+                No Pearl Safe
+              </Text>
+            )}
```

Comment:
I think we should display here all safes across chains but not only for selected agent homeChainId. the addresses should be the same across chains, but might be different historically when we had issues. could you verify with Roman please?